src/Lykke.AlgoStore.Api/Infrastructure/Extensions/HttpContextExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Linq;
+using Lykke.AlgoStore.Core.Utils;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+
+namespace Lykke.AlgoStore.Api.Infrastructure.Extensions
+{
+    public static class HttpContextExtensions
+    {
+        public static T GetHeaderValueAs<T>(this HttpContext httpContext, string headerName)
+        {
+            StringValues values;
+
+            if (httpContext?.Request?.Headers?.TryGetValue(headerName, out values) ?? false)
+            {
+                string rawValues = values.ToString();   // writes out as Csv when there are multiple.
+
+                if (!string.IsNullOrEmpty(rawValues))
+                    return (T)Convert.ChangeType(values.ToString(), typeof(T));
+            }
+            return default(T);
+        }
+        public static string GetIp(this HttpContext context)
```

Comment:
This extension method already exist in Lykke.Common, Version=4.1.0.0 Lykke.Common.Extensions.HttpContextExtensions